RobotEV3/src/main/java/de/fhg/iais/roberta/components/ev3c4ev3/C4Ev3SourceCompiler.java

```diff
@@ -15,14 +15,23 @@
     private final String compilerResourcesDir;
     private final String compilerExecutableFileName;
     private final String staticLibraryFolderName;
+    private final String compilerBinaryName;
 
-    public C4Ev3SourceCompiler(String compilerResourcesDir) {
+    public C4Ev3SourceCompiler(String compilerResourcesDir, String compilerBinaryName) {
         this.compilerResourcesDir = compilerResourcesDir;
-        this.compilerExecutableFileName = getCompilerExecutableFileName();
+        this.compilerExecutableFileName = compilerBinaryName;
         this.staticLibraryFolderName = getStaticLibraryFolderName();
+        this.compilerBinaryName = compilerBinaryName;
+    }
+
+    private String getCompilerBinaryName()
+    {
+        return  compilerBinaryName;
     }
 
     private String getCompilerExecutableFileName() {
```

Comment:
You need to fully replace the functionality of this method. Right now only Linux will work, as only that is entered as `robot.plugin.compiler.executable `. In order to solve this you can add a generic helper method somewhere, e.g. the `PluginProperties` which checks for `robot.plugin.compiler.executable` and some "sub"-properties like `robot.plugin.compiler.executable.windows`, `robot.plugin.compiler.executable.arm`, etc..
`robot.plugin.compiler.executable` should be used for all platforms in case no specific one exists.
That way you can specify the "default" one as just `executable` and override it with the platform specific ones if needed.

Reply:
Hi! Sorry for the delay! 
I made the changes you asked, tested them by running the server locally and compiling c4ev3 NEPO code.
I have in mind what the unit test needs to check, but i don't know in which package i should put it. Should i make a new test package?

Reply:
You can probably extract the functionality into a utility class and put the test in the util package.